glib/src/gstring.rs

```diff
@@ -505,32 +503,65 @@ impl StaticType for GString {
     }
 }
 
+impl crate::value::ValueType for GString {
+    type Type = String;
+}
+
+unsafe impl<'a> crate::value::FromValue<'a> for GString {
+    type Error = crate::value::WrongValueTypeOrNoneError<Self>;
+
+    unsafe fn from_value(value: &'a crate::Value) -> Self {
+        GString::from(<&str>::from_value(value))
+    }
+}
+
+impl crate::value::ToValue for GString {
+    fn to_value(&self) -> crate::Value {
+        <&str>::to_value(&self.as_str())
+    }
+
+    fn to_value_type(&self) -> Type {
+        String::static_type()
+    }
+}
+
+impl crate::value::ToValueOptional for GString {
```

Comment:
Why not the equivalent of `Into` and `TryInto` here? Like `ToValue` and `TryToValue`? That would seem more logical no?

Reply:
It's not `Into` because it's taking a reference instead of consuming the value, and it's not `Try` because it can't fail but instead it allows turning an `Option<T>` (that is: an optional) into a value.

A previous version of `FromValue` could've been a `TryFromValue` (it previously returned a `Result`), but the current version can't fail anymore either (and is `unsafe` because of that and the checking is done elsewhere, because otherwise there would be quite some code duplication for the checking).